src/GraphQL.Authorization.AspNetCore/GraphQLAuthExtensions.cs

```diff
@@ -9,7 +9,7 @@ public static class GraphQLAuthExtensions
     {
         public static void AddGraphQLAuth(this IServiceCollection services)
         {
-            services.TryAddSingleton<IAuthorizationEvaluator, AuthorizationEvaluator>();
+            services.TryAddScoped<IAuthorizationEvaluator, AuthorizationEvaluator>();
```

Comment:
Because `IAuthorizationService` is inject as `singleton`, I want to follow this convention. So you can change the code as:
```csharp
public static void AddGraphQLAuth(this IServiceCollection services, ServiceLifetime lifetime = ServiceLifetime.Singleton)
        {
            var authorizationEvaluatorDescriptor = ServiceDescriptor.Describe(typeof(IAuthorizationEvaluator), typeof(AuthorizationEvaluator), lifetime);
            services.TryAdd(authorizationEvaluatorDescriptor);

            services.AddTransient<IValidationRule, AuthorizationValidationRule>();
        }
```